src/realm/sync/network/default_socket.cpp

```diff
@@ -431,9 +431,15 @@ void DefaultWebSocketImpl::handle_ssl_handshake(std::error_code ec)
     if (ec) {
         REALM_ASSERT(ec != util::error::operation_aborted);
         constexpr bool was_clean = false;
-        websocket_error_and_close_handler(
-            was_clean,
-            Status{make_error_code(WebSocketError::websocket_tls_handshake_failed), ec.message()}); // Throws
+        std::error_code ec2;
+        if (ec == network::ssl::Errors::certificate_rejected) {
+            ec2 = make_error_code(WebSocketError::websocket_tls_handshake_failed);
```

Comment:
Do we want to just make this error non-fatal? Should [this](https://github.com/realm/realm-core/blob/f370b8060d55965bbfe0c7abe5b5b2bb3bfa3a77/src/realm/sync/noinst/client_impl_base.cpp#L548) be set to false? Wouldn't that have the same effect as this change?

Reply:
It would resolve the crash in the audit realm pool, but the error code/message would be misleading since the failure was due to a connection issue and not a TLS error.

Reply:
FYI - This is an example how the error currently looks in the sync logs:
```
HWOW[749:4098883] Sync: Connected to endpoint '65.0.217.240:443' (from '192.168.1.118:64482')
HWOW[749:4098883] Sync: Connection[1]: Closing the websocket with status='SystemError: record overflow', was_clean='false'
HWOW[749:4098883] Sync: Connection[1]: Connection closed due to error
HWOW[749:4098883] Sync: Events: Received sync error: SSL server certificate rejected: record overflow (ec=117)
```

Reply:
Should we do both?

Reply:
Sure - although I think the point of the fatal error in this case was to use the longer hold off time for a "true" SSL cert failure so the app doesn't keep hitting the server too quickly if it's going to keep failing anyways.

Honestly, if we see this error, it would be because the server's certificate had expired and the server team would need to fix it ASAP. Once we start using Tiered sync, however, this error might become more prominent.

Reply:
I think if the error code is tls_handshake_failed, you'll end up waiting an hour no matter what https://github.com/realm/realm-core/blob/f370b8060d55965bbfe0c7abe5b5b2bb3bfa3a77/src/realm/sync/noinst/client_impl_base.cpp#L687